### DISCUSSION TITLE
New or custom keyboard shortcuts?

### DISCUSSION THREAD

**Discussion by wilhelmloof** [mifi/lossless-cut#1646] [Ideas]
LosslessCut is an impressive media cutting tool. I wonder if it was partly inspired by the abandoned [MPEG Streamclip](http://www.squared5.com/svideo/mpeg-streamclip-mac.html), that also can losslessly cut videos? There is no available 64 bit version of MPEG Streamclip, so the last supported macOS version is Mojave. 

MPEG Streamclip has some very handy and intuitive keyboard shortcuts that I miss in LosslessCut.

**Example of keyboard shortcuts in MPEG Streamclip**

```
Right     = Next frame
Left      = Previous frame
Down      = Next keyframe
Up        = Previous keyframe
I         = Set in  (same as in LosslessCut)
O         = Set out (same as in LosslessCut)
Shift + I = Go to in
Shift + O = Go to out
X         = Cancel selection
CMD + T   = Trim to selection
CMD + I   = Show stream info
```

Would it be possible to implement some of these keyboard shortcuts in LosslessCut? Maybe as a custom set of shortcuts, if they are conflicting with present shortcuts.

**Comment by mifi** [comment]
Hi. Losslesscut already supports custom keyboard shortcuts. check out the Settings page. or am i misunderstanding your request?

**Reply by wilhelmloof** [reply]
Ah! My bad, I was using an older version. I didn't know that the changes where that big. Great that there are so many possible keyboard shortcut settings.

I was able find most of them, LosslessCut nomenclature in square brackets:

```
.           = Next frame      [Seeking/Step forward 1 frame]
,           = Previous frame  [Seeking/Step backward 1 frame]
Alt + right = Next keyframe      [Seeking/Seek next keyframe]
Alt + left  = Previous keyframe  [Seeking/Seek previous keyframe]
I           = Set in  [Segments and cut points/Start current segment at current time]
O           = Set out [Segments and cut points/End current segment at current time]
Shift + I   = Go to in  [Seeking/Jump to current segment's start time]
Shift + O   = Go to out [Seeking/Jump to current segment's end time]
Backspace   = Cancel selection [Segments and cut points/Remove current segment]
```

I was not able to find keyboard shortcut settings for `Trim to selection` (it might not fit LosslessCut's desired workflow) and `Show stream info`.

The list with shortcuts settings in LosslessCut is quite long, it would be handy with a filter/search box.

Where and in which format are the keyboard shortcut settings stored?

**Reply by mifi** [reply]
> I was not able to find keyboard shortcut settings for `Trim to selection` (it might not fit LosslessCut's desired workflow) and `Show stream info`.

what would  `Trim to selection` and `Show stream info` actions do?

> The list with shortcuts settings in LosslessCut is quite long, it would be handy with a filter/search box.

good idea. next version will have search for shortcuts.

> Where and in which format are the keyboard shortcut settings stored?

settings are stored in a json file in appData, see https://www.electronjs.org/docs/latest/api/app#appgetpathname

**Comment by wilhelmloof** [comment]
`Trim to selection` in MPEG Streamclip trims or you could say crops the media to the in and out points of active selection. After trim, the cropped media takes up the whole timeline. Thus if you are working with very long media, you can first make a rough selection, trim it and then use the whole timeline to make the fine selection.
Another way to accomplish this would be some kind of double timeline or a magnified timeline for the fine selection. Maybe by pressing Alt + moving the in or out marker, makes the whole timeline temporarily only show let's say -/+ 5% of the time around that marker. 

`Show stream info` is basically the most essential data from [MediaInfo](https://mediaarea.net/MediaInfo).

Thanks for adding a filter function to keyboard shortcut settings.

**Reply by mifi** [reply]
> `Trim to selection` in MPEG Streamclip trims or you could say crops the media to the in and out points of active selection. After trim, the cropped media takes up the whole timeline. Thus if you are working with very long media, you can first make a rough selection, trim it and then use the whole timeline to make the fine selection. Another way to accomplish this would be some kind of double timeline or a magnified timeline for the fine selection. Maybe by pressing Alt + moving the in or out marker, makes the whole timeline temporarily only show let's say -/+ 5% of the time around that marker.

not sure how this would work in losslesscut, because losslesscut doesn't have in/out markers, instead it has 1 timeline marker and multiple segments.


> `Show stream info` is basically the most essential data from [MediaInfo](https://mediaarea.net/MediaInfo).

isn't it the same as the existing `Edit tracks / metadata tags` shortcut?

**Reply by wilhelmloof** [reply]
`Show stream info` in MPEG Streamclip just shows basic data and has no possibilities to edit. Having 'Show' under Edit in the top menu is unintuitive. It's more logical to place Show media data under File or View.

It would also be nice to have some basic data optionally shown in the main interface. Like: resolution (width x height), container format, format full name, format profile, frame rate, codecs, sampling rate, channels, bitrates, etc. Although this is not critical, as it exists many tools to find media info.

[Invisor](https://www.invisorapp.com/) might be a good inspiration for media info.

**Reply by mifi** [reply]
> `Show stream info` in MPEG Streamclip just shows basic data and has no possibilities to edit. Having 'Show' under Edit in the top menu is unintuitive. It's more logical to place Show media data under File or View.

yes, but it's also used to edit, not only show. Not sure if putting it both under the View and the Edit menu would be less confusing.

I agree the tracks and file metadata UI could use some improvements. i added suggestion here: #1649
